src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/form-control.source.ts

```diff
@@ -2,11 +2,11 @@ import { SetPollingSourceToolipsTexts } from "src/app/common/tooltips/tooltips.t
 import { FetchKind, MergeKind, ReadKind } from "./add-polling-source-form.types";
 
 export interface RadioControlType {
-    id: string;
+    id?: string;
     value: string;
-    icon: string;
+    icon?: string;
     label: string;
-    tooltip: string;
+    tooltip?: string;
```

Comment:
Why did you make these potentially undefined?
This looks like a very dirty hack.

Reply:
Yes, returned it as it was before